### DISCUSSION TITLE
[Feature] record http status codes >= 300

### DISCUSSION THREAD

**Discussion by h0cheung** [apache/skywalking#12411] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

As for now, all http status codes < 400 will not be recorded.

#5327 was closed because status code 200 is usually not of interest to users.

However, status codes such as 301, 302, etc., which imply redirections, belong to a different class than 2xx, and users may be interested in them.

Maybe we should record status >= 300 instead of status >= 400

### Related issues

#5327

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Sorry to say, different people have different use cases.

3xx, e.g. 302, is not a special case for many people. If you want to build metrics for it, please fo ahead to use OAL, and aggregate at OAP side.

**Reply by h0cheung** [reply]
As I understand it, the input data of the OAL seems to be spans, right? The 3xx status code isn't recorded in spans.
